hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -1537,6 +1640,277 @@ public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
+			final var accountStore = createAccountStore();
+			final var tokenStore = createTokenStore(accountStore, sideEffectsTracker);
+			final var payerAccount = accountStore.loadAccount(Id.fromGrpcAccount(EntityIdUtils.accountIdFromEvmAddress(senderAddress)));
+
+			/* --- Execute the transaction and capture its results --- */
+			final var precompileAddress = Address.fromHexString(HTS_PRECOMPILED_CONTRACT_ADDRESS);
+
+			if (isFungible) {
+				final var checkResponseCode = allowanceChecks.allowancesValidation(transactionBody.getCryptoApproveAllowance().getCryptoAllowancesList(),
+						transactionBody.getCryptoApproveAllowance().getTokenAllowancesList(),
+						transactionBody.getCryptoApproveAllowance().getNftAllowancesList(),
+						payerAccount,
+						currentView);
+
+				validateTrue(checkResponseCode == OK, checkResponseCode);
+
+				final var approveAllowanceLogic = approveAllowanceLogicFactory.newApproveAllowanceLogic(
+						accountStore, tokenStore, dynamicProperties);
+				approveAllowanceLogic.approveAllowance(transactionBody.getCryptoApproveAllowance().getCryptoAllowancesList(),
+						transactionBody.getCryptoApproveAllowance().getTokenAllowancesList(),
+						transactionBody.getCryptoApproveAllowance().getNftAllowancesList(),
+						EntityIdUtils.accountIdFromEvmAddress(frame.getSenderAddress()));
+				frame.addLog(getLogForFungibleAdjustAllowance(precompileAddress));
+			} else {
+				if (approveOp.spender().getAccountNum() == 0) {
+					final var deleteAllowanceLogic = deleteAllowanceLogicFactory.newDeleteAllowanceLogic(accountStore, tokenStore);
+					deleteAllowanceLogic.deleteAllowance(transactionBody.getCryptoDeleteAllowance().getNftAllowancesList(),
+							EntityIdUtils.accountIdFromEvmAddress(frame.getSenderAddress()));
+				} else {
+					final var checkResponseCode = allowanceChecks.allowancesValidation(transactionBody.getCryptoApproveAllowance().getCryptoAllowancesList(),
+							transactionBody.getCryptoApproveAllowance().getTokenAllowancesList(),
+							transactionBody.getCryptoApproveAllowance().getNftAllowancesList(),
+							payerAccount,
+							currentView);
+
+					validateTrue(OK.equals(checkResponseCode), checkResponseCode);
+
+					final var approveAllowanceLogic = approveAllowanceLogicFactory.newApproveAllowanceLogic(
+							accountStore, tokenStore, dynamicProperties);
+					approveAllowanceLogic.approveAllowance(transactionBody.getCryptoApproveAllowance().getCryptoAllowancesList(),
+							transactionBody.getCryptoApproveAllowance().getTokenAllowancesList(),
+							transactionBody.getCryptoApproveAllowance().getNftAllowancesList(),
+							EntityIdUtils.accountIdFromEvmAddress(frame.getSenderAddress()));
+				}
+				frame.addLog(getLogForNftAdjustAllowance(precompileAddress));
+			}
+		}
+
+		@Override
+		public long getMinimumFeeInTinybars(Timestamp consensusTime) {
+			Objects.requireNonNull(approveOp);
+
+			return precompilePricingUtils.getMinimumPriceInTinybars(
```

Comment:
This can also be a `CryptoDeleteAllowance`, right? We should add it to `GasCostType` and handle it accordingly